store/reducer.ts

```diff
@@ -0,0 +1,24 @@
+import { ADD_PHOTO, addPhotoAction } from './actions';
+import { MainState } from './types';
+
+const initialState: MainState = {
+  photoData: [],
+};
+
+export const reducer = (state = initialState, action: addPhotoAction): MainState => {
```

Comment:
As it is a common reducer for the app, I suggest creating a type `StateAction` which can be `addPhotoAction`